openmdao/core/_checks.py

```diff
@@ -112,11 +112,6 @@ def __check_shapes_match(src, target, to_prom_name):
                                  _both_names(src, to_prom_name),
                                  target['size'],
                                  _both_names(target, to_prom_name)))
-            if len(src_idxs) > src['size']:
-                raise ValueError("Size %d of target indices is larger than size"
-                                 " %d of source %s." %
-                                 (len(src_idxs), src['size'],
-                                 _both_names(src, to_prom_name)))
```

Comment:
We talked about having a check for max(src_idxs) < src['size'].  Did you find we had that check already somewhere else?